src/azure-cli/azure/cli/command_modules/storage/_client_factory.py

```diff
@@ -376,3 +376,37 @@ def cf_table_service(cli_ctx, kwargs):
 
 def cf_table_client(cli_ctx, kwargs):
     return cf_table_service(cli_ctx, kwargs).get_table_client(table_name=kwargs.pop('table_name'))
+
+
+def cf_share_service(cli_ctx, kwargs):
+    client_kwargs = prepare_client_kwargs_track2(cli_ctx)
+    client_kwargs = _config_location_mode(kwargs, client_kwargs)
+    t_share_service = get_sdk(cli_ctx, ResourceType.DATA_STORAGE_FILESHARE, '_share_service_client#ShareServiceClient')
+    connection_string = kwargs.pop('connection_string', None)
+    account_key = kwargs.pop('account_key', None)
+    token_credential = kwargs.pop('token_credential', None)
+    sas_token = kwargs.pop('sas_token', None)
+    account_name = kwargs.pop('account_name', None)
+    account_url = kwargs.pop('account_url', None)
+    if connection_string:
+        return t_share_service.from_connection_string(conn_str=connection_string, **client_kwargs)
+    if not account_url:
+        account_url = get_account_url(cli_ctx, account_name=account_name, service='file')
+    credential = account_key or sas_token or token_credential
+
+    if account_url and credential:
+        return t_share_service(account_url=account_url, credential=credential, **client_kwargs)
+    return None
```

Comment:
```suggestion
    return t_share_service(account_url=account_url, credential=credential, **client_kwargs)
```
We can't allow `account_url` or `credential` is None